WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -385,75 +385,19 @@ - (void)showNotificationsTab
-                                                                         andTags:[options stringForKey:WPNewPostURLParamTagsKey]
-                                                                        andImage:[options stringForKey:WPNewPostURLParamImageKey]];
-            }
-            
-            if (options[WPPostViewControllerOptionNotAnimated]) {
-                animated = NO;
-            }
-        }
-        navController = [[UINavigationController alloc] initWithRootViewController:editPostViewController];
-        navController.restorationIdentifier = WPEditorNavigationRestorationID;
-        navController.restorationClass = [WPPostViewController class];
-    } else {
-        WPLegacyEditPostViewController *editPostLegacyViewController;
-        if (!options) {
-            editPostLegacyViewController = [[WPLegacyEditPostViewController alloc] initWithDraftForLastUsedBlog];
-            NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
-            BlogService *blogService = [[BlogService alloc] initWithManagedObjectContext:context];
-            Blog *blog = [blogService lastUsedOrFirstBlog];
-            [WPAppAnalytics track:WPAnalyticsStatEditorCreatedPost withProperties:@{ @"tap_source": @"tab_bar"} withBlog:blog];
-        } else {
-            editPostLegacyViewController = [[WPLegacyEditPostViewController alloc] initWithTitle:[options stringForKey:WPNewPostURLParamTitleKey]
-                                                                                      andContent:[options stringForKey:WPNewPostURLParamContentKey]
-                                                                                         andTags:[options stringForKey:WPNewPostURLParamTagsKey]
-                                                                                        andImage:[options stringForKey:WPNewPostURLParamImageKey]];
-        }
-        navController = [[UINavigationController alloc] initWithRootViewController:editPostLegacyViewController];
-        navController.restorationIdentifier = WPLegacyEditorNavigationRestorationID;
-        navController.restorationClass = [WPLegacyEditPostViewController class];
-    }
-
-    navController.modalPresentationStyle = UIModalPresentationFullScreen;
-    navController.navigationBar.translucent = NO;    
-    [self presentViewController:navController animated:animated completion:^{
-        WPImpactFeedbackGenerator *generator = [[WPImpactFeedbackGenerator alloc] initWithStyle:WPImpactFeedbackStyleMedium];
-        [generator impactOccurred];
-    }];
+    EditPostViewController* editor = [EditPostViewController new];
+    editor.modalPresentationStyle = UIModalPresentationFullScreen;
+    editor.showImmediately = !animated;
+    editor.openWithMediaPicker = openToMedia;
+    [WPAppAnalytics track:WPAnalyticsStatEditorCreatedPost withProperties:@{ @"tap_source": @"tab_bar"} withBlog:editor.blog];
```

Comment:
I mentioned this above, but I think the `EditPostViewController` does this same analytics post?